nifi-nar-bundles/nifi-kite-bundle/nifi-kite-processors/src/main/java/org/apache/nifi/processors/kite/InferAvroSchemaFromCSV.java

```diff
@@ -0,0 +1,235 @@
+            .name("CSV Header Line")
+            .description("Comma separated string defining the column names expected in the CSV data. " +
+                    "EX: \"fname,lname,zip,address\"")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor HEADER_LINE_SKIP_COUNT = new PropertyDescriptor.Builder()
+            .name("CSV Header Line Skip Count")
+            .description("Specifies the number of header lines that should be skipped when reading the CSV data. If the " +
+                    " first line of the CSV data is a header line and you specify the \"CSV Header Line\" property " +
+                    "you need to set this vlaue to 1 otherwise the header line will be treated as actual data.")
+            .required(true)
+            .defaultValue("0")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor ESCAPE_STRING = new PropertyDescriptor.Builder()
+            .name("CSV escape string")
+            .description("String that represents an escape sequence in the CSV FlowFile content data.")
+            .required(true)
+            .defaultValue("\\")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor QUOTE_STRING = new PropertyDescriptor.Builder()
+            .name("CSV quote string")
+            .description("String that represents a literal quote character in the CSV FlowFile content data.")
+            .required(true)
+            .defaultValue("'")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor RECORD_NAME = new PropertyDescriptor.Builder()
+            .name("Avro Record Name")
+            .description("Value to be placed in the Avro record schema \"name\" field.")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor CHARSET = new PropertyDescriptor.Builder()
+            .name("Charset")
+            .description("Character encoding of CSV data.")
+            .required(true)
+            .defaultValue("UTF-8")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor PRETTY_AVRO_OUTPUT = new PropertyDescriptor.Builder()
+            .name("Pretty Avro Output")
+            .description("If true the Avro output will be formatted.")
+            .required(true)
+            .defaultValue("true")
+            .allowableValues("true", "false")
+            .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .build();
+
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder().name("success")
```

Comment:
Does it make sense to call this relationship 'schema' rather than 'success'? Seems like it may be more intuitive, but I'll leave that determination up to you, if you prefer 'success'.